mantidproject/errorreports
# Add command RemoveOldReports

- A new command has been added to "Remove Old Reports" actually it removes the user identifiable info from the reports and deletes UserDetails that are not tied to and Active report (one that hasn't had it's user info removed, as it occured in the last 90 days). The `id` attached to old ErrorReport hasn't been deleted, but I believe this corresponds to the name and email of UserDetails which is getting deleted!
- This was based off of a [command file to remove old recovery files](https://github.com/mantidproject/errorreports/blob/master/web/services/management/commands/removeolddata.py) (which I have replaced as removing recovery files is useless now we no longer accept recovery files!)
- Also, when I launched docker there was a migration file (from another PR: https://github.com/mantidproject/errorreports/commit/38584f03ea1606f55dcfec87e2e9efb3bd4ef621) which had not been logged in this repo, so I've included it here.

@martyngigg  I assume a Cron job can be run on the actual server along the lines of 
`0 0 * * * docker exec errorreports_web_1 python manage.py removeoldreports`

To test:

On a local server (with mantid.properties: errorreports.rooturl = http://localhost:8082).
Then create some error reports! I created about 5 reports, waited 3 minutes and created more. 
You will need to edit the command file (web/services/management/commands/removeoldreports.py) to filter for 5 minutes rather than 90 days. To do this, change line 26 and 48 to: `...timedelta(**minutes=options['days'])**)`

To actually delete user info from the old reports, (assuming the server is already up!) you need to run: 
`docker exec errorreports_web_1 python manage.py removeoldreports`
This defaults to run the "days" command (which would usually delete reports older than 90 days, but you've set this to 90 mins!), if you add the `--all` argument, then as expected, all error reports will be deleted. 


You probably want to run: `docker exec errorreports_web_1 python manage.py removeoldreports 5` to delete reports older than 5 mins.
By the time you have shutdown and re-booted the local server, the first handful of reports will be <5 mins old and the later ones will still be in the last 5 mins! Good luck with your timings!

To create error reports: load `TrainingCourseData/Training_Exercise3a_SNS.nxs`, executing `NormaliseToMonitor` algorithm and clicking "Yes, Share Info"
Repeat as necessary!
